### ISSUE TITLE
Enable important modules after D7 upgrade is complete

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#6451]
## Description of the need

There are issues in the D2B Migrate issue queue to enable some modules after the upgrade is complete. Specifically:

- Project Browser (https://github.com/backdrop-contrib/d2b_migrate/issues/46)
- Dashboard (https://github.com/backdrop-contrib/d2b_migrate/issues/51)

## Alternatives that have been considered

Leaving this for the D2B module to handle.

## Additional information

It was my experience with the D2B module, that after upgrade there is a Dashboard link in menu, but the page is empty because the module is not enabled. If this is also true of the core upgrade process, then this feels a bit broken.

**Comment by argiepiano** [comment]
We do have the ability to enable these automatically during upgrade. In fact this was discussed in #5506, which was prompted by #5499. It was decided not to enable Dashboard, Project Installers and others. May it's time to rethink that decision.

**Comment by jenlampton** [comment]
I think its important that we not add functionality to anyone's site as part of the upgrade. If they didn't have it before, they won't be expecting it afterward. Plus, this could cause security issues if previously untrusted users all of a sudden get permission to install modules (for example). 

However, I do think we should do a better job of providing people who have upgraded with information that these new features are available so that they have the option to turn them on if they like. Some places we might want to put this information:
* Info message immediately after completing the upgrade
* Status report (but only for those who upgraded)
* Dashboard (will only work if Dashboard was a module they had enabled in Drupal 7) -- but  think there is a core bug for this right now anyway. 
* Other ideas?

**Comment by kiamlaluno** [comment]
When a site is migrated from Drupal 7, is the Dashboard module enabled?  
What prompted this issue was the Dashboard link that took to an empty page, in a site that migrated using the d2b_migrate module.

**Comment by stpaultim** [comment]
> I think its important that we not add functionality to anyone's site as part of the upgrade. If they didn't have it before, they won't be expecting it afterward.

I have mixed feelings about this statement. There are quite a few "out of the box" improvements that folks who upgrade from Drupal 7 experience when they move to Backdrop CMS. The layouts system provides a lot of functionality that didn't exist in the previous site. I understand the layout system is integral to Backdrop and not something that we could easily exclude from upgrades. My point is that I think new functionality after an upgrade is both inevitable and mostly desireable.

When I talk to people about upgrading, I get the sense that people do expect new functionality when they upgrade their site and are a bit disappointed if they don't see new features and functionality - after engaging in the time and expense to upgrade.

Having said that, I understand that this is not always the case and feel as if we should approach this on a case by case basis. Informing people before the upgrade of what is available to them is one strategy, another is making them aware of what has changed. 

As @kiamlaluno pointed out, I raised the issue of the dashboard mostly because in my recent upgrade the "dashboard" link was there, but it was broken, because the dashboard module was not enabled. This is a bug and makes Backdrop CMS look broken. Maybe it was unique to my upgrade, but I think I've seen this before. I would appreciate hearing if other people have experienced this. 

We could solve this by either making sure the dashboard link is not available or by enabling the dashboard module. It seems to me that the dashboard doesn't really "add functionality" to the site. Rather, it's a new UI tool to expose existing functionality to site managers. We added it because we thought it would improve peoples experience on new sites, I don't see any reason to assume this is less true for existing sites upgrading to Backdrop CMS. 

I suggested enabling the project browser module, mostly because I've heard that request from other people while we've been experimenting with the D2B Migrate module. I see that this could be a bit more problematic, security wise, but it seems as if there should be solutions to enable doing this in a secure way. 

One option would be to provide site upgraders with the option to enable certain "recommended" modules after the upgrade, so they have the option to accept or decline. I think that would be sufficient for both the dashboard module and the project browser. I don't know how easy this would be. But, I don't think that we are not doing anyone a favor by effectively "hiding" some important features after upgrade. 

At a minimum, let's fix the broken "Dashboard" link after upgrade, unless it turns out that this was specific to my site.

**Comment by laryn** [comment]
> in my recent upgrade the "dashboard" link was there, but it was broken, because the dashboard module was not enabled.

I think this may be caused by the D7 Dashboard module (a completely different module) being left enabled before the upgrade. Then during/after the upgrade Backdrop thinks its own Dashboard module is enabled, but none of the install process has run, leading to this breakage.

**Comment by argiepiano** [comment]
> I think this may be caused by the D7 Dashboard module (a completely different module) being left enabled before the upgrade

Exactly. The [upgrade documentation even specifies](https://docs.backdropcms.org/documentation/upgrading-drupal-7-modules-that-have-been-removed-from-core) that, quote:

Dashboard module:
- Disable and uninstall the module
- After the upgrade: enable Dashboard module (different module, same name)

**Comment by stpaultim** [comment]
I checked and the Dashboard module was not enabled on the Drupal 7 site in question. However, I have yet to verify that the problem is repeatable. It seems like I've seen this before after other upgrades, but I don't have solid documenation yet. At this point, my report is still anecdotal.

**Comment by olafgrabienski** [comment]
> It seems like I've seen this before after other upgrades, but I don't have solid documenation yet.

There is a related issue with some observations: https://github.com/backdrop/backdrop-issues/issues/5529

**Comment by kiamlaluno** [comment]
The other issue says there is no Dashboard link in the admin bar. [quicksketch](https://github.com/backdrop/backdrop-issues/issues/5529#issuecomment-1213560769) then says `admin/dashboard` exists, but the admin bar links are screwed up.

What said here about the Dashboard link seems a third variant of the that issue.

**Comment by stpaultim** [comment]
I opened this issue after being triggered by a specific bug, but suggested a solution that is bigger than necessary to fix the bug. I've opened a new issue focused on that specific bug in the upgrade process. 
https://github.com/backdrop/backdrop-issues/issues/6457
